assets/js/modules/tagmanager/components/common/WebContainerNameTextField.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WebContainerNameTextField component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { isURL } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME, FORM_SETUP, CONTAINER_CREATE } from '../../datastore/constants';
+import { STORE_NAME as CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { STORE_NAME as CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+import ContainerNameTextField from './ContainerNameTextField';
+const { useSelect, useDispatch } = Data;
+
+export default function WebContainerNameTextField() {
+	const containerID = useSelect( ( select ) => select( STORE_NAME ).getContainerID() );
+	const siteName = useSelect( ( select ) => select( CORE_SITE ).getSiteName() );
+	const isSecondaryAMP = useSelect( ( select ) => select( CORE_SITE ).isSecondaryAMP() );
+	const referenceSiteURL = useSelect( ( select ) => select( CORE_SITE ).getReferenceSiteURL() );
+
+	let containerName = siteName;
+	if ( ! containerName && isURL( referenceSiteURL ) ) {
+		containerName = new URL( referenceSiteURL ).hostname;
+	}
+
+	const { setValues } = useDispatch( CORE_FORMS );
+	useEffect( () => {
+		setValues( FORM_SETUP, { containerName } );
```

Comment:
We should only call `setValues` here if the `containerName` is empty.

Because this happens unconditionally, the storybook stories for the warning don't show it because this effect is likely overriding it on render.

Reply:
Ah... my bad... I got it now :doh:... Updated.